File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task164_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

std::vector<std::string> lastNameLenSort(std::vector<std::string> names) {
    std::sort(names.begin(), names.end(), [](const std::string& a, const std::string& b) {
        auto aLastSpace = a.find_last_of(\' \');
        auto bLastSpace = b.find_last_of(\' \');
        std::string aLastName = a.substr(aLastSpace + 1);
        std::string bLastName = b.substr(bLastSpace + 1);
        
        if (aLastName.length() != bLastName.length()) {
            return aLastName.length() < bLastName.length();
        }
        return aLastName < bLastName;
    });
    return names;
}

int main() {
    std::vector<std::string> names = {
        "Jennifer Figueroa",
        "Heather Mcgee",
        "Amanda Schwartz",
        "Nicole Yoder",
        "Melissa Hoffman"
    };
    
    auto sorted = lastNameLenSort(names);
    
    for (const auto& name : sorted) {
        std::cout << name << std::endl;
    }
    
    return 0;
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task164_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.*;

public class NameSorter {
    public static List<String> lastNameLenSort(List<String> names) {
        names.sort((a, b) -> {
            String[] aParts = a.split(" ");
            String[] bParts = b.split(" ");
            String aLastName = aParts[aParts.length - 1];
            String bLastName = bParts[bParts.length - 1];
            
            if (aLastName.length() != bLastName.length()) {
                return Integer.compare(aLastName.length(), bLastName.length());
            }
            return aLastName.compareTo(bLastName);
        });
        return names;
    }

    public static void main(String[] args) {
        List<String> names = Arrays.asList(
            "Jennifer Figueroa",
            "Heather Mcgee",
            "Amanda Schwartz",
            "Nicole Yoder",
            "Melissa Hoffman"
        );
        System.out.println(lastNameLenSort(names));
    }
}
